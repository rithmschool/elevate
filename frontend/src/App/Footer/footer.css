.footer-brand-name {
  font-size: 1.5em;
  font-family: "Exo 2", sans-serif;
  color: #394557;
  opacity: 0.65;
  letter-spacing: 3px;
}

.footer-brand-name:hover {
  color: #394557;
  opacity: 0.85;
}

.link {
  font-weight: bold;
  font-size: 14px;
  color: #394557;
  padding: 0px 0px;
  margin: 0px 0px;
  display: inline-block;
  position: relative;
  opacity: 0.75;
}

.link:hover {
  opacity: 1;
  color: #394557;
}

.link::before {
  transition: 300ms;
  height: 2px;
  content: "";
  position: absolute;
  background-color: #394557;
}

.link-ltr::before {
  width: 0%;
  bottom: 0px;
}

.link-ltr:hover::before {
  width: 100%;
}

.footer-container {
    background-color: #dfe7ed;
    border: 1px #dcd9d5 solid;
    align-items: center;
    justify-content: center;
    margin-top: 5px;
    width: 100vw;
    height: 40vh;
    padding: 8% 0;
}