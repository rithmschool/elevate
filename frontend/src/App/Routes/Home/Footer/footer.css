.Footer {
  border-top: 1px #dcd9d5 solid;
  padding: 2rem 0;
  margin: .5rem;
}

.Footer a {
  display: inline-block;
  font-weight: bold;
  font-size: 14px;
  color: rgba(57, 69, 87, 0.75);
}

.Footer a:hover {
  opacity: 1;
  color: #394557;
}

.Footer a::before {
  transition: 300ms;
  height: 2px;
  content: "";
  position: absolute;
  background-color: #394557;
}

.Footer a.brand-logo {
  font-weight: normal;
  font-size: 1.5em;
  font-family: "Exo 2", sans-serif;
  color: #6558f5;
  opacity: 0.65;
  letter-spacing: 3px;
}

.Footer a.brand-logo:hover {
  color: #6558f5;
  opacity: 0.85;
}

.Footer button {
  background-color: #6558f5;
}
