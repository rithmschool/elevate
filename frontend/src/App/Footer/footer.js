import React from "react";
import {
  Button,
  Container,
  Row,
  Col,
  Form,
  FormGroup,
  Label,
  Input
} from "reactstrap";
import { Link } from "react-router-dom";
import './footer.css';

class Footer extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      email: ""
    };
  }

  handleChange = evt => {
    this.setState({ [evt.target.name]: evt.target.value });
  };

  render() {
    return (
      <div className="footer-container">
        <Container>
          <Row>
            <Col>
              <h6>Stay up to date with Brella</h6>
              <Form inline>
                <FormGroup>
                  <Label for="newsletter-email" hidden>
                    newsletter email
                  </Label>
                  <Input
                    type="text"
                    id="newsletter-email"
                    name="email"
                    placeholder="Email Address"
                    value={this.state.email}
                    onChange={this.handleChange}
                  />
                </FormGroup>
                <Button>Sign Up</Button>
              </Form>
            </Col>
            <Col>
              <Link className="link link-ltr" to="/ask-an-expert">Blog</Link>
              <br></br>
              <Link className="link link-ltr" to="/ask-an-expert">Contact</Link>
              <br></br>
              <Link className="link link-ltr" to="/ask-an-expert">Facebook</Link>
              <br></br>
              <Link className="link link-ltr" to="/ask-an-expert">Instagram</Link>
              <br></br>
              <Link className="link link-ltr" to="/ask-an-expert">Twitter</Link>
              <br></br>
            </Col>
            <Col>
              <Link className="footer-brand-name" to="/">
                Elevate
              </Link>
            </Col>
          </Row>
        </Container>
      </div>
    );
  }
}

export default Footer;
