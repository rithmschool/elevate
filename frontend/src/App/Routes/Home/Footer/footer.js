import React from "react";
import {
  Container,
  Row,
  Col,
  Form,
  FormGroup,
  Label,
  Input
} from "reactstrap";
import { Link } from "react-router-dom";
import "./footer.css";
import ElevateApi from "../../../../elevateApi";

class Footer extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      email: "",
      signedUp: false
    };
  }

  handleChange = evt => {
    this.setState({ [evt.target.name]: evt.target.value });
  };

  handleSubmit = async evt => {
    evt.preventDefault();
    await ElevateApi.postNewsletterSignUp(this.state);
    this.setState({
      email: "",
      signedUp: true
    });
  };

  render() {
    return (
      <div className="Footer">
        <Container>
          <Row>
            <Col className="d-flex flex-column justify-content-center">
              {this.state.signedUp ? (
                <h6>Thanks for signing up!</h6>
              ) : (
                <div>
                  <h6>Stay up to date with Brella</h6>
                  <Form onSubmit={this.handleSubmit}>
                    <FormGroup>
                      <Label for="email" hidden>
                        newsletter email
                      </Label>
                      <Input
                        type="text"
                        id="email"
                        name="email"
                        placeholder="Email Address"
                        value={this.state.email}
                        onChange={this.handleChange}
                      />
                    </FormGroup>
                  </Form>
                </div>
              )}
            </Col>
            <Col>
              <Link to="/ask-an-expert">Blog</Link>
              <br></br>
              <Link to="/ask-an-expert">Contact</Link>
              <br></br>
              <Link to="/ask-an-expert">Facebook</Link>
              <br></br>
              <Link to="/ask-an-expert">Instagram</Link>
              <br></br>
              <Link to="/ask-an-expert">Twitter</Link>
              <br></br>
            </Col>
            <Col className="d-flex align-items-center justify-content-end">
              <Link className="brand-logo" to="/">
                Brella
              </Link>
            </Col>
          </Row>
        </Container>
      </div>
    );
  }
}

export default Footer;
