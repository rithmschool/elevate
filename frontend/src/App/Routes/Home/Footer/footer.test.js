import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Footer from "./footer";
import { MemoryRouter } from "react-router-dom";

describe("Footer", function() {
  let wrapper;

  it('renders', () => {
    wrapper = shallow(<Footer />);
  });

  beforeEach(() => {
    var onSubmitFn = jest.fn();
    wrapper = mount(
      <MemoryRouter>
        <Footer onSubmit={onSubmitFn}/>
      </MemoryRouter>
    );
  });

  it("matches snapshot", function() {
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });

  it("changes signedUp state on submit", function() {
    var onSubmitFn = jest.fn();
    wrapper = mount(
      <MemoryRouter>
        <Footer onSubmit={onSubmitFn}/>
      </MemoryRouter>
    );
    wrapper.setState({
      email: "",
      signedUp: false
    });
    console.log("wrapper state", wrapper.state())
    let form = wrapper.find("form");
    form.simulate("submit");
    // console.log("new wrapper state", wrapper.state())
    // expect(wrapper.state("signedUp")).toEqual(true);
    expect(onSubmitFn).toHaveBeenCalledTimes(1);

    wrapper.update();

    expect(wrapper.find('input[id="email"]')).toHaveLength(0);
  });
});
