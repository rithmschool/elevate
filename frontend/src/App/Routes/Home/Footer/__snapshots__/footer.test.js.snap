// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Footer>
  <div
    className="Footer"
  >
    <Container
      fluid={true}
    >
      <div
        className="container-fluid"
      >
        <ForwardRef
          noGutters={false}
        >
          <div
            className="row"
          >
            <Col
              className="d-flex flex-column justify-content-center"
            >
              <div
                className="d-flex flex-column justify-content-center col"
              >
                <newsletterSignUpForm>
                  <div>
                    <div>
                      <h6>
                        Stay up to date with Brella
                      </h6>
                      <Form
                        onSubmit={[Function]}
                        tag="form"
                      >
                        <form
                          className=""
                          onSubmit={[Function]}
                        >
                          <FormGroup
                            tag="div"
                          >
                            <div
                              className="form-group"
                            >
                              <Label
                                for="email"
                                hidden={true}
                                tag="label"
                                widths={
                                  Array [
                                    "xs",
                                    "sm",
                                    "md",
                                    "lg",
                                    "xl",
                                  ]
                                }
                              >
                                <label
                                  className="sr-only"
                                  htmlFor="email"
                                >
                                  newsletter email
                                </label>
                              </Label>
                              <Input
                                id="email"
                                name="email"
                                onChange={[Function]}
                                placeholder="Email Address"
                                type="text"
                                value=""
                              >
                                <input
                                  className="form-control"
                                  id="email"
                                  name="email"
                                  onChange={[Function]}
                                  placeholder="Email Address"
                                  type="text"
                                  value=""
                                />
                              </Input>
                            </div>
                          </FormGroup>
                        </form>
                      </Form>
                    </div>
                  </div>
                </newsletterSignUpForm>
              </div>
            </Col>
            <Col
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                }
              }
            >
              <div
                className="col"
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": "column",
                  }
                }
              >
                <Link
                  to="/ask-an-expert"
                >
                  <LinkAnchor
                    href="/ask-an-expert"
                    navigate={[Function]}
                  >
                    <a
                      href="/ask-an-expert"
                      onClick={[Function]}
                    >
                      Blog
                    </a>
                  </LinkAnchor>
                </Link>
                <Link
                  to="/ask-an-expert"
                >
                  <LinkAnchor
                    href="/ask-an-expert"
                    navigate={[Function]}
                  >
                    <a
                      href="/ask-an-expert"
                      onClick={[Function]}
                    >
                      Contact
                    </a>
                  </LinkAnchor>
                </Link>
                <Link
                  to="/ask-an-expert"
                >
                  <LinkAnchor
                    href="/ask-an-expert"
                    navigate={[Function]}
                  >
                    <a
                      href="/ask-an-expert"
                      onClick={[Function]}
                    >
                      Facebook
                    </a>
                  </LinkAnchor>
                </Link>
                <Link
                  to="/ask-an-expert"
                >
                  <LinkAnchor
                    href="/ask-an-expert"
                    navigate={[Function]}
                  >
                    <a
                      href="/ask-an-expert"
                      onClick={[Function]}
                    >
                      Instagram
                    </a>
                  </LinkAnchor>
                </Link>
                <Link
                  to="/ask-an-expert"
                >
                  <LinkAnchor
                    href="/ask-an-expert"
                    navigate={[Function]}
                  >
                    <a
                      href="/ask-an-expert"
                      onClick={[Function]}
                    >
                      Twitter
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
            </Col>
            <Col
              className="d-flex align-items-center justify-content-end"
            >
              <div
                className="d-flex align-items-center justify-content-end col"
              >
                <Link
                  className="brand-logo"
                  to="/"
                >
                  <LinkAnchor
                    className="brand-logo"
                    href="/"
                    navigate={[Function]}
                  >
                    <a
                      className="brand-logo"
                      href="/"
                      onClick={[Function]}
                    >
                      Brella
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
            </Col>
          </div>
        </ForwardRef>
      </div>
    </Container>
  </div>
</Footer>
`;
